[2024-09-13 00:35:55,577] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
/workspace/Oobleck/oobleck/__init__.py
node_index: 8
[2024-09-13 00:35:58,891] [WARNING] [agent.py:154:_launch_workers] Profile data for model gpt2 not found. Exception Error parsing json file: /workspace/Oobleck/tmp/profiles/gpt3_2_7B-8-15-1/mb8.json Launching profiler...
[2024-09-13 00:36:00,847] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-09-13 00:36:03,955] [INFO] [profiler.py:270:profile] Profiling model gpt2. rank 8, world_size 15, num_workers_per_node 1.
Dataset: model_name: gpt2, dataset_path: wikitext, dataset_name: wikitext-2-raw-v1, max_seq_length: 1024
[2024-09-13 00:36:06,708] [INFO] [profiler.py:276:profile] Dataset loaded
model_name: gpt2, model_tag: gpt3_2_7B
[2024-09-13 00:36:07,897] [INFO] [profiler.py:284:profile] model loaded
[2024-09-13 00:36:47,105] [INFO] [profiler.py:292:profile] init pg. master_addr: 172.21.0.42, master_port: 23456
[2024-09-13 00:36:51,299] [INFO] [profiler.py:306:profile] Profiling model execution latency.
[2024-09-13 00:36:51,301] [INFO] [profiler.py:112:profile_execution_layers] before barrier
[2024-09-13 00:37:04,112] [INFO] [profiler.py:114:profile_execution_layers] after barrier
[2024-09-13 00:37:04,113] [INFO] [profiler.py:120:profile_execution_layers] before broadcast
[2024-09-13 00:37:04,113] [INFO] [profiler.py:122:profile_execution_layers] after broadcast
[2024-09-13 00:37:04,117] [INFO] [profiler.py:317:profile] Profiling cross-node allreduce latency.
[2024-09-13 00:37:04,117] [INFO] [profiler.py:160:profile_allreduce_across_nodes] Profile allreduce across 15 nodes latency
[2024-09-13 00:37:04,411] [INFO] [profiler.py:326:profile] Profiling in-node allreduce latency.
[2024-09-13 00:37:04,411] [INFO] [profiler.py:209:profile_allreduce_in_node] Profile allreduce within a node latency
[2024-09-13 00:37:05,568] [INFO] [agent.py:189:_launch_workers] in agent. my_ip_port 172.21.0.47:2220, node_ip_ports ['172.21.0.42:2220', '172.21.0.42:2221', '172.21.0.42:2222', '172.21.0.42:2223', '172.21.0.46:2220', '172.21.0.46:2221', '172.21.0.46:2222', '172.21.0.46:2223', '172.21.0.47:2220', '172.21.0.47:2221', '172.21.0.47:2222', '172.21.0.47:2223', '172.21.0.90:2220', '172.21.0.90:2221', '172.21.0.90:2222']
[2024-09-13 00:37:05,568] [INFO] [agent.py:192:_launch_workers] Launching worker 0...
[2024-09-13 00:37:07,501] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-09-13 00:37:10,751] [INFO] [worker.py:23:worker_main] Initializing Oobleck Engine...
[2024-09-13 00:37:10,751] [INFO] [worker.py:24:worker_main] in worker main: my_ip 172.21.0.47:2220
/root/miniconda3/envs/oobleck/lib/python3.10/site-packages/transformers/training_args.py:1281: FutureWarning: using `no_cuda` is deprecated and will be removed in version 5.0 of ðŸ¤— Transformers. Use `use_cpu` instead
  warnings.warn(
Dataset: model_name: gpt2, dataset_path: wikitext, dataset_name: wikitext-2-raw-v1, max_seq_length: 1024
[2024-09-13 00:37:13,527] [INFO] [engine.py:635:_initialize_engine] model arguments: {'n_embd': 2560, 'n_head': 32, 'n_positions': 1024, 'num_hidden_layers': 32}
model_name: gpt2, model_tag: gpt3_2_7B
mb size: 34, ar across: 34, ar in: 34
Loading done. creating layer execution results...
Returning from get_profiler_results
[2024-09-13 00:37:14,721] [INFO] [engine.py:677:_initialize_engine] Number of nodes range: (3, 15)
min_nodes: 3, max_nodes: 15, gpus: 1, 
Create pipeline templates
StageExecutionResult[0:12] with 1 devices
StageExecutionResult[13:23] with 1 devices
StageExecutionResult[24:33] with 1 devices
StageExecutionResult[0:9] with 1 devices
StageExecutionResult[10:18] with 1 devices
StageExecutionResult[19:26] with 1 devices
StageExecutionResult[27:33] with 1 devices
StageExecutionResult[0:7] with 1 devices
StageExecutionResult[8:14] with 1 devices
StageExecutionResult[15:21] with 1 devices
StageExecutionResult[22:27] with 1 devices
StageExecutionResult[28:33] with 1 devices
StageExecutionResult[0:6] with 1 devices
StageExecutionResult[7:12] with 1 devices
StageExecutionResult[13:18] with 1 devices
StageExecutionResult[19:23] with 1 devices
StageExecutionResult[24:28] with 1 devices
StageExecutionResult[29:33] with 1 devices
StageExecutionResult[0:5] with 1 devices
StageExecutionResult[6:10] with 1 devices
StageExecutionResult[11:15] with 1 devices
StageExecutionResult[16:20] with 1 devices
StageExecutionResult[21:25] with 1 devices
StageExecutionResult[26:29] with 1 devices
StageExecutionResult[30:33] with 1 devices
StageExecutionResult[0:5] with 1 devices
StageExecutionResult[6:10] with 1 devices
StageExecutionResult[11:14] with 1 devices
StageExecutionResult[15:18] with 1 devices
StageExecutionResult[19:22] with 1 devices
StageExecutionResult[23:26] with 1 devices
StageExecutionResult[27:30] with 1 devices
StageExecutionResult[31:33] with 1 devices
StageExecutionResult[0:4] with 1 devices
StageExecutionResult[5:8] with 1 devices
StageExecutionResult[9:12] with 1 devices
StageExecutionResult[13:16] with 1 devices
StageExecutionResult[17:20] with 1 devices
StageExecutionResult[21:24] with 1 devices
StageExecutionResult[25:27] with 1 devices
StageExecutionResult[28:31] with 1 devices
StageExecutionResult[32:33] with 1 devices
StageExecutionResult[0:4] with 1 devices
StageExecutionResult[5:8] with 1 devices
StageExecutionResult[9:12] with 1 devices
StageExecutionResult[13:16] with 1 devices
StageExecutionResult[17:19] with 1 devices
StageExecutionResult[20:22] with 1 devices
StageExecutionResult[23:25] with 1 devices
StageExecutionResult[26:28] with 1 devices
StageExecutionResult[29:31] with 1 devices
StageExecutionResult[32:33] with 1 devices
StageExecutionResult[0:4] with 1 devices
StageExecutionResult[5:7] with 1 devices
StageExecutionResult[8:10] with 1 devices
StageExecutionResult[11:13] with 1 devices
StageExecutionResult[14:16] with 1 devices
StageExecutionResult[17:19] with 1 devices
StageExecutionResult[20:22] with 1 devices
StageExecutionResult[23:25] with 1 devices
StageExecutionResult[26:28] with 1 devices
StageExecutionResult[29:31] with 1 devices
StageExecutionResult[32:33] with 1 devices
StageExecutionResult[0:3] with 1 devices
StageExecutionResult[4:6] with 1 devices
StageExecutionResult[7:9] with 1 devices
StageExecutionResult[10:12] with 1 devices
StageExecutionResult[13:15] with 1 devices
StageExecutionResult[16:18] with 1 devices
StageExecutionResult[19:21] with 1 devices
StageExecutionResult[22:24] with 1 devices
StageExecutionResult[25:26] with 1 devices
StageExecutionResult[27:28] with 1 devices
StageExecutionResult[29:31] with 1 devices
StageExecutionResult[32:33] with 1 devices
StageExecutionResult[0:3] with 1 devices
StageExecutionResult[4:6] with 1 devices
StageExecutionResult[7:9] with 1 devices
StageExecutionResult[10:12] with 1 devices
StageExecutionResult[13:15] with 1 devices
StageExecutionResult[16:18] with 1 devices
StageExecutionResult[19:21] with 1 devices
StageExecutionResult[22:24] with 1 devices
StageExecutionResult[25:26] with 1 devices
StageExecutionResult[27:28] with 1 devices
StageExecutionResult[29:30] with 1 devices
StageExecutionResult[31:32] with 1 devices
StageExecutionResult[33:33] with 1 devices
StageExecutionResult[0:3] with 1 devices
StageExecutionResult[4:6] with 1 devices
StageExecutionResult[7:9] with 1 devices
StageExecutionResult[10:12] with 1 devices
StageExecutionResult[13:15] with 1 devices
StageExecutionResult[16:17] with 1 devices
StageExecutionResult[18:19] with 1 devices
StageExecutionResult[20:21] with 1 devices
StageExecutionResult[22:23] with 1 devices
StageExecutionResult[24:25] with 1 devices
StageExecutionResult[26:27] with 1 devices
StageExecutionResult[28:29] with 1 devices
StageExecutionResult[30:31] with 1 devices
StageExecutionResult[32:33] with 1 devices
[2024-09-13 00:37:31,222] [DEBUG] [agent.py:298:on_receive_response] Receiving: (<Response.FORWARD_RANK0_PORT: 5>, <RequestType.UNDEFINED: 0>)
[2024-09-13 00:37:31,223] [DEBUG] [agent.py:243:on_receive_worker_port] agent recv TCP Store port 36823 from master
StageExecutionResult[0:3] with 1 devices
StageExecutionResult[4:6] with 1 devices
StageExecutionResult[7:9] with 1 devices
StageExecutionResult[10:12] with 1 devices
StageExecutionResult[13:14] with 1 devices
StageExecutionResult[15:16] with 1 devices
StageExecutionResult[17:18] with 1 devices
StageExecutionResult[19:20] with 1 devices
StageExecutionResult[21:22] with 1 devices
StageExecutionResult[23:24] with 1 devices
StageExecutionResult[25:26] with 1 devices
StageExecutionResult[27:28] with 1 devices
StageExecutionResult[29:30] with 1 devices
StageExecutionResult[31:32] with 1 devices
StageExecutionResult[33:33] with 1 devices
[2024-09-13 00:37:31,480] [INFO] [worker.py:27:worker_main] Initializing torch.distributed...
[2024-09-13 00:37:31,481] [DEBUG] [engine.py:725:initialize_distributed] regenerate rank_map {'172.21.0.42:2220': [0], '172.21.0.42:2221': [1], '172.21.0.42:2222': [2], '172.21.0.42:2223': [3], '172.21.0.46:2220': [4], '172.21.0.46:2221': [5], '172.21.0.46:2222': [6], '172.21.0.46:2223': [7], '172.21.0.47:2220': [8], '172.21.0.47:2221': [9], '172.21.0.47:2222': [10], '172.21.0.47:2223': [11], '172.21.0.90:2220': [12], '172.21.0.90:2221': [13], '172.21.0.90:2222': [14]}
[2024-09-13 00:37:31,481] [INFO] [engine.py:739:initialize_distributed] init pg: rank 8, world_size: 15, rank_map: {'172.21.0.42:2220': [0], '172.21.0.42:2221': [1], '172.21.0.42:2222': [2], '172.21.0.42:2223': [3], '172.21.0.46:2220': [4], '172.21.0.46:2221': [5], '172.21.0.46:2222': [6], '172.21.0.46:2223': [7], '172.21.0.47:2220': [8], '172.21.0.47:2221': [9], '172.21.0.47:2222': [10], '172.21.0.47:2223': [11], '172.21.0.90:2220': [12], '172.21.0.90:2221': [13], '172.21.0.90:2222': [14]}
[2024-09-13 00:37:31,481] [INFO] [engine.py:755:initialize_distributed] Waiting for a port information...
[2024-09-13 00:37:31,481] [INFO] [engine.py:758:initialize_distributed] Received torch master: 172.21.0.42:2220.36823
[2024-09-13 00:37:31,638] [INFO] [comm.py:637:init_distributed] cdb=None
[2024-09-13 00:37:31,638] [INFO] [engine.py:775:initialize_distributed] [rank: 8] Distributed initialization is done.
[2024-09-13 00:37:31,638] [INFO] [worker.py:34:worker_main] Instantiating pipelines...
MindtPy initialization may have generated poor quality cuts.
[2024-09-13 00:37:31,750] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:37:32,018] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4942E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:37:32,284] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
MindtPy initialization may have generated poor quality cuts.
[2024-09-13 00:37:32,338] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:37:32,568] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value 2.5059E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:37:33,782] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:37:34,098] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4942E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:37:34,438] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4942E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:37:34,810] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:37:35,994] [INFO] [instantiator.py:382:_distribute_batch] Batch distribution find no results. return None
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
The objective value -7.4941E-09 of feasibility problem is less than zero_tolerance. This indicates that the nlp subproblem is feasible, although it is found infeasible in the previous step. Check the nlp solver output
[2024-09-13 00:37:39,506] [INFO] [instantiator.py:256:get_best_execution_plan] Best execution plan: 1 x <oobleck.PipelineTemplate.4nodes> pipelines (b: 34)
1 x <oobleck.PipelineTemplate.5nodes> pipelines (b: 43)
1 x <oobleck.PipelineTemplate.6nodes> pipelines (b: 51)
B: 128
dataloader: mbs 8
[2024-09-13 00:37:55,024] [INFO] [engine.py:159:_reconfiguration_listener_fn] ReconfigureEngine: start reconfigure listening
[2024-09-13 00:37:55,025] [INFO] [worker.py:36:worker_main] Begin training...
step: 0
step in try: 0
load_mbs: 0.0006320476531982422, forward time: 9.636979818344116, backward time: 21.11661195755005
load_mbs cnt: 43, forward cnt: 43, backward cnt: 43
training step: 0 done
/root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/cuda/memory.py:416: FutureWarning: torch.cuda.memory_cached has been renamed to torch.cuda.memory_reserved
  warnings.warn(
/root/miniconda3/envs/oobleck/lib/python3.10/site-packages/torch/cuda/memory.py:424: FutureWarning: torch.cuda.max_memory_cached has been renamed to torch.cuda.max_memory_reserved
  warnings.warn(
step: 1
step in try: 1
load_mbs: 0.0006113052368164062, forward time: 9.204695224761963, backward time: 21.124916076660156
load_mbs cnt: 43, forward cnt: 43, backward cnt: 43
training step: 1 done
step: 2
step in try: 2
load_mbs: 0.0005865097045898438, forward time: 9.205116748809814, backward time: 21.124426126480103
load_mbs cnt: 43, forward cnt: 43, backward cnt: 43
training step: 2 done
step: 3
step in try: 3
load_mbs: 0.0005772113800048828, forward time: 9.20376992225647, backward time: 21.123762845993042
load_mbs cnt: 43, forward cnt: 43, backward cnt: 43
training step: 3 done
step: 4
step in try: 4
load_mbs: 0.00058746337890625, forward time: 9.202993869781494, backward time: 21.12427306175232
load_mbs cnt: 43, forward cnt: 43, backward cnt: 43
training step: 4 done
step: 5
step in try: 5
load_mbs: 0.0005862712860107422, forward time: 9.202867269515991, backward time: 21.12285041809082
load_mbs cnt: 43, forward cnt: 43, backward cnt: 43
training step: 5 done
step: 6
step in try: 6
load_mbs: 0.0005958080291748047, forward time: 9.204152822494507, backward time: 21.124702215194702
load_mbs cnt: 43, forward cnt: 43, backward cnt: 43
training step: 6 done
step: 7
step in try: 7
load_mbs: 0.0006284713745117188, forward time: 9.20447325706482, backward time: 21.123814582824707
load_mbs cnt: 43, forward cnt: 43, backward cnt: 43
training step: 7 done
step: 8
step in try: 8
load_mbs: 0.0006303787231445312, forward time: 9.204331159591675, backward time: 21.122994661331177
load_mbs cnt: 43, forward cnt: 43, backward cnt: 43
training step: 8 done
step: 9
step in try: 9
load_mbs: 0.0005991458892822266, forward time: 9.202457427978516, backward time: 21.12286877632141
load_mbs cnt: 43, forward cnt: 43, backward cnt: 43
training step: 9 done
[2024-09-13 00:51:43,844] [INFO] [engine.py:932:train] Training is done. Waiting for synchronization...
[2024-09-13 00:51:44,448] [INFO] [engine.py:935:train] Training done.
